Pokemon/Pokemon/PokemonTableViewController.m

```diff
@@ -0,0 +1,358 @@
+    NSArray *flying = @[ @"pigeot", @"pidgeotto", @"pidgey", @"spearow", @"zapdos", @"zubat"];
+    NSArray *poison = @[ @"arbok", @"beedrill", @"ekans", @"golbat", @"grimer", @"koffing", @"muk", @"nidoking", @"nidoqueen", @"nidoran-f", @"nidoran", @"nidorina", @"nidorino", @"tentacool", @"tentacruel", @"weezing"];
+    NSArray *ice = @[ @"articuno", @"jynx"];
+    NSArray *bug = @[@"beedrill", @"caterpie", @"butterfree", @"kakuna", @"metapod", @"paras", @"parasect", @"pinsir", @"scyther", @"venomoth", @"venonat", @"weedle"];
+    NSArray *normal = @[@"chansey", @"ditto", @"dodrio", @"doduo", @"eevee", @"farfetchd", @"fearow", @"jigglypuff", @"kangaskhan", @"lickitung", @"meowth", @"persian", @"porygon", @"rattatta", @"snorlax", @"tauros"];
+    NSArray *fairy = @[@"clefable", @"wigglytuff"];
+    NSArray *ground = @[@"cubone", @"diglett", @"dugtrio", @"marowak", @"rhydon", @"rhyhorn", @"sandshrew", @"sandslash"];
+    NSArray *dragon = @[@"dragonair", @"dragonite", @"dratini"];
+    NSArray *electric = @[@"electabuzz", @"electrode", @"jolteon", @"magnemite", @"magneton", @"pikachu", @"raichu", @"voltorb"];
+    NSArray *ghost = @[@"gastly", @"gengar", @"haunter"];
+    NSArray *fighting = @[@"hitmonchan", @"machamp", @"machoke", @"machop", @"mankey", @"primeape"];
+    
+    
+    self.tableData = @{@"fire" : fire, @"water" : water, @"grass" : grass, @"psychic" : psychic, @"rock" : rock, @"flying" : flying, @"poison" : poison, @"ice" : ice, @"bug" : bug, @"normal" : normal, @"fairy" : fairy, @"ground" : ground, @"dragon" : dragon, @"electric" : electric, @"ghost" : ghost, @"fighting" : fighting };
+}
+
+- (void)didReceiveMemoryWarning {
+    [super didReceiveMemoryWarning];
+    // Dispose of any resources that can be recreated.
+}
+
+
+-(void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender
+{
+-(IBAction)segmentbutton:(id)sender {
+      NSIndexPath *indexPath = [self.tableView indexPathForSelectedRow];
+    if (segmentController.selectedSegmentIndex == 0) {
+        NSString * pokemonName = self.allPokemon[indexPath.row];
+        PokemonDetailViewController *vc = segue.destinationViewController;
+        
+        vc.pokemonName = pokemonName;
+        }
+    else {
+        NSArray * keys = [self.tableData allKeys];
+        NSString *key = keys[indexPath.section];
+        NSArray * pokemonArray = [self.tableData objectForKey:key];
+        NSString * pokemonName = pokemonArray[indexPath.row];
+        PokemonDetailViewController *vc = segue.destinationViewController;
+        
+        vc.pokemonName = pokemonName;
+        
+        }
+}
+    
+
+#pragma mark - Table view data source
+
+
+- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
+    // Return the number of sections.
+    
+    
+-(IBAction)segmentbutton:(id)sender {
+    if (segmentController.selectedSegmentIndex == 0) {
+        return 1;
+    }
+    else {
+          return [self.tableData allKeys].count;
+    }
```

Comment:
Having two `}`s indented at the same level should be a red flag that you've got some mis-matched braces somewhere.